### PR TITLE
UncommittedGC fix for azure support

### LINKED ISSUE #4844: Discovery: GC+ support for Azure
2 days discovery to check GC+ on Azure, if it's not working what is missing to support it.

### PR DESCRIPTION
**Pull request by nopcoder**
Close #4844

Changes to support Azure part of discovery - tested manually.

### CODE REVIEW

File: clients/spark/core/src/main/scala/io/treeverse/gc/UncommittedAddressLister.scala [hunk 1]
```diff
@@ -23,10 +23,10 @@ class APIUncommittedAddressLister(apiClient: ApiClient) extends UncommittedAddre
     var continuationToken: String = null
     do {
       resp = apiClient.prepareGarbageCollectionUncommitted(repo, continuationToken)
-      continuationToken = resp.getContinuationToken()
+      continuationToken = resp.getContinuationToken
```
Comment (N-o-Z) [inline clients/spark/core/src/main/scala/io/treeverse/gc/UncommittedAddressLister.scala:26]: These changes seem unnecessary?

File: clients/spark/core/src/main/scala/io/treeverse/gc/UncommittedGarbageCollector.scala [hunk 2]
```diff
@@ -156,7 +157,11 @@ object UncommittedGarbageCollector {
 
         var uncommittedDF =
           if (uncommittedGCRunInfo.uncommittedLocation != "") {
-            spark.read.parquet(uncommittedGCRunInfo.uncommittedLocation)
+            val uncommittedLocation =
```
Comment (N-o-Z) [inline clients/spark/core/src/main/scala/io/treeverse/gc/UncommittedGarbageCollector.scala:160]: Can you explain what exactly was the issue with Azure?

**Review by N-o-Z — CHANGES_REQUESTED** [review]
Thanks - some comments

File: clients/spark/core/src/main/scala/io/treeverse/gc/UncommittedGarbageCollector.scala (hunk 2 above)
Comment (nopcoder) [reply clients/spark/core/src/main/scala/io/treeverse/gc/UncommittedGarbageCollector.scala:160]: When we use Hadoop's FileSystem for the repository storage namespace we are getting AbstractFileSystem that can't be used for listing.
By converting the storage namespace address to `abfs://...` scheme and query for FileSystem, we are getting AzureBlobStoreFileSystem that enables the API we use.
Used the same method that the current GC users - fetch the right storage NS address based on the client you use.

File: clients/spark/core/src/main/scala/io/treeverse/gc/UncommittedAddressLister.scala (hunk 1 above)
Comment (nopcoder) [reply clients/spark/core/src/main/scala/io/treeverse/gc/UncommittedAddressLister.scala:26]: Just my IDE's scala recommended way calling java accessor methods:

image

**Review by N-o-Z — APPROVED** [review]
Thanks

**Pull request merged**